ddtrace/internal/telemetry/telemetry_writer.py

```diff
@@ -0,0 +1,184 @@
+                resp = get_connection_response(conn)
+
+                t = sw.elapsed()
+                if t >= self.interval:
+                    log_level = logging.WARNING
+                else:
+                    log_level = logging.DEBUG
+
+                log.log(log_level, "sent %d in %.5fs to %s. response: %s", len(rb_json), t, self.url, resp.status)
+            finally:
+                conn.close()
+        return resp
+
+    def flush_integrations_queue(self):
+        # type () -> List[Dict]
+        """Returns a list of all integrations queued by classmethods"""
+        with self._lock:
+            integrations = self._integrations_queue
+            self._integrations_queue = []
+        return integrations
+
+    def flush_requests_queue(self):
+        # type () -> List[TelemetryRequest]
+        """Returns a list of all integrations queued by classmethods"""
+        with self._lock:
+            requests = self._requests_queue
+            self._requests_queue = []
+        return requests
+
+    def periodic(self):
+        integrations = self.flush_integrations_queue()
+        if integrations:
+            integrations_request = app_integrations_changed_telemetry_request(integrations)
+            TelemetryWriter.add_request(integrations_request)
+
+        requests = self.flush_requests_queue()
+        requests_failed = []  # type: List[TelemetryRequest]
+        for request in requests:
+            resp = self._send_request(request)
+            if resp.status >= 300:
+                requests_failed.append(request)
+
+        with self._lock:
+            # add failed requests back to the requests queue
+            self._requests_queue.extend(requests_failed)
+
+    def shutdown(self):
+        # type: () -> None
+        appclosed_request = app_closed_telemetry_request()
+        self.add_request(appclosed_request)
+        self.periodic()
+
+    @classmethod
+    def add_request(cls, request):
+        # type: (TelemetryRequest) -> None
+        """
+        Adds a Telemetry Request to the TelemetryWriter request buffer
+
+         :param TelemetryRequest request: dictionary which stores a formatted telemetry request body and header
+        """
+        with cls._lock:
+            if cls._instance is None:
+                return
+            request["body"]["seq_id"] = cls.sequence
+            cls.sequence += 1
+            # TO DO: replace list with a dead letter queue
+            cls._instance._requests_queue.append(request)
+
+    @classmethod
+    def integration_event(cls, integration_name):
+        # type: (str) -> None
+        """
+        Creates and queues a the name and configurations of a patched module
+
+        :param str integration_name: name of patched module
+        """
+        with cls._lock:
+            if cls._instance is None:
+                return
+            integration = create_integration(integration_name)
```

Comment:
we probably don't need this `create_integration` helper function, since it just creates a dict we can inline that creation to this function.